### ISSUE TITLE
No mapstore docker image for 20.1.x series & future 22.0

### ISSUE THREAD

**Issue by fvanderbiest** [georchestra/georchestra#3199]
https://github.com/georchestra/docker/blob/56023e53f420ef6948cf8cd70ffd83ec1856ffc9/docker-compose.yml#L154
https://hub.docker.com/r/georchestra/mapstore/tags/

Should it be imported into the main repository ?

**Comment by fvanderbiest** [comment]
We now have a stable tag (1.1.x):
https://hub.docker.com/r/geosolutionsit/mapstore2-georchestra/tags?page=1&ordering=last_updated

We should probably update the composition to use it in stable branches

**Comment by fvanderbiest** [comment]
> We should probably update the composition to use it in stable branches

Done with https://github.com/georchestra/docker/commit/98cd1a316ea55159c620e617aa567a8ed07b0515

**Comment by fvanderbiest** [comment]
But now we have `geosolutionsit/mapstore2-georchestra:1.1.x` on `20.1` and `georchestra/mapstore:latest` on `master`.
This will require some homogeneisation

**Comment by fvanderbiest** [comment]
Reopening since, ideally, we should:
 * import the mapstore app into the main repo, 
 * revisit the dockerfile to use jetty, 
 * create a github action
 * publish a `georchestra/mapstore:20.1.x` docker image

**Comment by fvanderbiest** [comment]
Discussions ongoing on this subject...
I'm more and more convinced that there's not future in trying to integrate all apps in the core repository, with docker images tagged homogeneously.
Instead it should be kept lightweight and geOrchestra should be able to use modules which live their own life in their repositories.
This is a big change.

**Comment by fvanderbiest** [comment]
As soon as a tag is pushed by GeoSolutions onto https://github.com/georchestra/mapstore2-georchestra then, according to https://github.com/georchestra/mapstore2-georchestra/blob/master/.github/workflows/mapstore.yml we'll have a docker image to use for the next release.

**Comment by pmauduit** [comment]
For now we only build master (latest) and tags. Do we also need to build the `2021.02.xx` branch (where tags like `20.1.5-georchestra` are set) ? Comparing to what buildbot does, we only have a build from the master branch there.

**Comment by landryb** [comment]
https://packages.georchestra.org/bot/#/builders/7 is a builder for ms2 stable branch on buildbot :)

**Comment by pmauduit** [comment]
ok, I missed this one :) looking into the master.cfg buildbot config, I figured out that only the master branch was spied on:

```
# mapstore2-georchestra build
c['change_source'].append(changes.GitPoller(
        repourl='https://github.com/georchestra/mapstore2-georchestra/',
        project='mapstore2-georchestra',
        workdir='workdir', branches=['master'],
        pollinterval=3600))

```

**Comment by landryb** [comment]
well, good catch, that was probably a bug :) fixed !

**Comment by pmauduit** [comment]
So now I think we are still missing a gh action following the branch `2021.02.xx`, which would take care of building docker images for 20.x series, but I wonder if it is still worth, considering the coming 22.0 release ?

**Comment by landryb** [comment]
afaict branch 2021.02.xx doesnt have adherence on master/22.x and should just work on 20.x series...

**Comment by jmbourdaret** [comment]
once merged, https://github.com/georchestra/mapstore2-georchestra/pull/497
should build `2021.02.xx` images as well,
one adherence that might exist between *geOr* versions and *mapstore* versions, is the mapstore datadir location, which introduced ``/etc/georchestra/mapstore/configs`` in "recent" mapstore versions but this was sorted out long ago.

(also, still on tomcat)

**Comment by pmauduit** [comment]
> once merged, [georchestra/mapstore2-georchestra#497](https://github.com/georchestra/mapstore2-georchestra/pull/497)

done

About tomcat vs jetty, I'd prefer to keep tomcat, as it was the initial choice from the development team.

**Comment by fvanderbiest** [comment]
Cool, we should be able to use `georchestra/mapstore:2021.02.xx` on future 22.0.x series if I understand correctly.

https://hub.docker.com/layers/georchestra/mapstore/2021.02.xx/images/sha256-18697b8d7c9ea9320abcd388380708d49f7aac469b376e3f13b3aeea69ee22a0?context=explore
